### PR TITLE
[Added][BEL-2924]Dangerfile PR in blocking mode

### PR DESCRIPTION
**Pull request by jaume-blv**
**What**
Added a workflow to validate PR in blocking mode to do not allow merge.

**Why**
- Force PR titles to contain a Jira ticket ID and a bracket indicating its purpose "Fixed|Added|Changed|Removed|Security|Other".
- Force the PR description to have at least 10 characters

### CODE REVIEW

**Review by jaume-blv — COMMENTED** [review]
LGTM

File: Dangerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+fail "Please provide a short summary in the PR description :page_with_curl:" if github.pr_body.length < 10
+
+# The title should include the correct prefix tag
+if !github.pr_title.match(/^\[(?:Fixed|Added|Changed|Removed|Security|Other)\]/)
+	fail "Please provide a valid PR title label: [Added/[Fixed]/[Changed]/[Removed]/[Security]/[Other]"
+end
+
+# The title should include the JIRA ticket unless is a dependabot PR
+if github.pr_labels.include?("dependencies")
+	message ("Autogenerated PR by renovate bot")
```
Comment (anisaoshafi) [inline Dangerfile:13]:
```suggestion
	message ("PR autogenerated by dependabot")
```

File: Dangerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Mainly to encourage writing up some reasoning about the PR, rather than
+# just leaving a title
+
```
Comment (anisaoshafi) [inline Dangerfile:3]:
```suggestion
```

File: Dangerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+fail "Please provide a short summary in the PR description :page_with_curl:" if github.pr_body.length < 10
+
+# The title should include the correct prefix tag
+if !github.pr_title.match(/^\[(?:Fixed|Added|Changed|Removed|Security|Other)\]/)
+	fail "Please provide a valid PR title label: [Added/[Fixed]/[Changed]/[Removed]/[Security]/[Other]"
```
Comment (anisaoshafi) [inline Dangerfile:8]:
```suggestion
	fail "Please provide a valid PR title label: [Added]/[Fixed]/[Changed]/[Removed]/[Security]/[Other]"
```

**Review by anisaoshafi — CHANGES_REQUESTED** [review]
Small comments

File: Dangerfile (hunk 3 above)
Comment (jaume-blv) [reply Dangerfile:8]: mmhhh nice, thanks

File: Dangerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# The title should include the correct prefix tag
+if !github.pr_title.match(/^\[(?:Fixed|Added|Changed|Removed|Security|Other)\]/)
+	fail "Please provide a valid PR title label: [Added]/[Fixed]/[Changed]/[Removed]/[Security]/[Other]"
+end
+
+# The title should include the JIRA ticket unless is a dependabot PR
+if github.pr_labels.include?("dependencies")
+	message ("PR autogenerated by dependabot")
+elsif !github.pr_title.match(/^\[.*\]\s?\[BEL-\d+\]/)
+	fail "Please provide a valid Jira ticket ID associated to this PR: [BEL-XXXX]. If you do not have any associated Jira ticket, just use [BEL-XXXX] "
```
Comment (anisaoshafi) [inline Dangerfile:14]: a nit pick:
```suggestion
	fail "Please provide a valid Jira ticket ID associated to this PR: [BEL-XXXX]. If you do not have any associated Jira ticket, just use [BEL-XXXX]"
```

Comment (jaume-blv) [reply Dangerfile:14]: 💋

**Review by anisaoshafi — APPROVED** [review]
lgtm 👍🏼

File: Dangerfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+# The title should include the correct prefix tag
+if !github.pr_title.match(/^\[(?:Fixed|Added|Changed|Removed|Security|Other)\]/)
+	fail "Please provide a valid PR title label: [Added]/[Fixed]/[Changed]/[Removed]/[Security]/[Other]"
+end
+
+# The title should include the JIRA ticket unless is a dependabot PR
+if github.pr_labels.include?("dependencies")
+	message ("PR autogenerated by dependabot")
+elsif !github.pr_title.match(/^\[.*\]\s?\[BEL-\d+\]/)
```
Comment (gingrassia) [inline Dangerfile:13]: this is a public repo, which means people from outside Belvo can create PRs... so we will be forcing them to add this even when they don't have one. What should we do there?

**Pull request merged**